Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs

```diff
@@ -77,23 +84,66 @@ protected override void InitializeModel()
             Input.OrderIndex = propertyOrderIndex++;
 
             Key.IsPrincipal = true;
+            Key.IsVisible = true;
             Key.OrderIndex = propertyOrderIndex++;
 
+            KeySecureString.IsPrincipal = true;
+            KeySecureString.IsVisible = false;
+            KeySecureString.OrderIndex = propertyOrderIndex++;
+
+            KeyInputModeSwitch.IsVisible = false;
+
             Encoding.IsPrincipal = false;
             Encoding.OrderIndex = propertyOrderIndex++;
             Encoding.Value = null;
             Encoding.IsRequired = true;
 
-            KeySecureString.IsPrincipal = false;
-            KeySecureString.OrderIndex = propertyOrderIndex++;
-
             Result.IsPrincipal = false;
             Result.OrderIndex = propertyOrderIndex++;
 
             ContinueOnError.IsPrincipal = false;
             ContinueOnError.OrderIndex = propertyOrderIndex++;
             ContinueOnError.Widget = new DefaultWidget { Type = ViewModelWidgetType.NullableBoolean };
             ContinueOnError.Value = false;
+
+            MenuActionsBuilder<KeyInputMode>.WithValueProperty(KeyInputModeSwitch)
+                .AddMenuProperty(Key, KeyInputMode.Key)
+                .AddMenuProperty(KeySecureString, KeyInputMode.SecureKey)
+                .BuildAndInsertMenuActions();
+        }
+
+        /// <inheritdoc/>
+        protected override void InitializeRules()
+        {
+            base.InitializeRules();
+            Rule(nameof(KeyInputModeSwitch), KeyInputModeChanged_Action);
+        }
+
+        /// <inheritdoc/>
+        protected override void ManualRegisterDependencies()
+        {
+            base.ManualRegisterDependencies();
+            RegisterDependency(KeyInputModeSwitch, nameof(KeyInputModeSwitch.Value), nameof(KeyInputModeSwitch));
+        }
+
+        /// <summary>
+        /// Files input Mode has changed. Set controls visibility based on selection
+        /// </summary>
+        private void KeyInputModeChanged_Action()
+        {
+            switch (KeyInputModeSwitch.Value)
+            {
+                case KeyInputMode.Key:
+                    Key.IsVisible = true;
+                    KeySecureString.IsVisible = false;
+                    break;
+                case KeyInputMode.SecureKey:
+                    KeySecureString.IsVisible = true;
+                    Key.IsVisible = false;
```

Comment:
keep properties in the same order (Key / SecureKey), as the other case (applicable to all activities)

Reply:
Done